aptos-move/framework/aptos-framework/sources/primary_store.move

```diff
@@ -0,0 +1,156 @@
+/// This defines the module for interacting with primary stores of accounts/objects, which have deterministic addresses
+module aptos_framework::primary_store {
+    use aptos_framework::fungible_asset::{Self, FungibleAsset, FungibleAssetStore};
+    use aptos_framework::object::{Self, Object, ConstructorRef, DeriveRef};
+
+    use std::signer;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Resource stored on the fungible asset metadata object to allow creating primary stores for it.
+    struct DeriveRefPod has key {
+        metadata_derive_ref: DeriveRef,
+    }
+
+    /// Creators of fungible assets can call this to enable support for creating primary (deterministic) stores for
+    /// their users.
+    public fun add_derived_ref_pod(
+        constructor_ref: &ConstructorRef,
+    ) {
+        let metadata_obj = &object::generate_signer(constructor_ref);
```

Comment:
Why did you remove the call to fungible_asset::add_fungibility?

Reply:
change it back.
the assumption changes after I make the derived_object function friend.